### PR TITLE
fix(css): make scrollbar visible again

### LINKED ISSUE #24619: Visibility of scroll bar in main UI
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**
I am having a problem on scrolling in both vertical and horizontal scrolls in Nextcloud UI. The color of the scroll and the width is too small. 

**Describe the solution you'd like**
Using visible color like dark gray or any visible color would fix this thing, and widen the scroll will be good for easy use of scrolls.

**Describe alternatives you've considered**

**Additional context**
Nextcloud Version: 20.0.2
Web Browser: Microsoft Edge Version 87.0.664.57
Thank you so much

### PR DESCRIPTION
**Pull request by ornanovitch**
* Resolves: #24619 

## Summary

Let the browser do its job regarding the appearance of the scrollbar

Sorry for the two unrelated changes, my neovim doesn't love trailing spaces :smile_cat: 

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Screenshots before/after for front-end changes (idk how to screenshot with the scrollbar always visible...)
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by susnux — COMMENTED** [review]
I agree that the current scrollbar is not very accessible.
So I fully support removing the `scrollbar-width`, but for the `scrollbar-color` I think we should use theming.
The color is broken it should be `var(--color-border-maxcontrast) transparent` and for our high-contrast themes we should set it to `auto`.

But would also be fine with removing the styles for the scrollbar if designers agree with that :)

### Before

[before.webm](https://github.com/user-attachments/assets/fe3ce091-5ffb-4437-82ec-efe00859f6b8)

### After (this PR)

[after.webm](https://github.com/user-attachments/assets/e454f131-58e0-4b72-b737-32f5f9d7dff0)

**Review by jancborchardt — COMMENTED** [review]
@susnux Good point with using color-text-maxcontrast for the scrollbar color. @Ornanovitch could you adjust that?

Otherwise it looks good. :)

File: core/css/styles.scss [hunk 1]
```diff
@@ -20,8 +20,6 @@ html, body, div, span, object, iframe, h1, h2, h3, h4, h5, h6, p, blockquote, pr
 	font-family: inherit;
 	vertical-align: baseline;
 	cursor: default;
```
Comment (susnux) [inline core/css/styles.scss:22]: So if we want to do it with theming:

```suggestion
	cursor: default;
	scrollbar-color: var(--scrollbar-color);
```

And in `apps/theming/lib/Themes/DefaultTheme.php` add to `$variables` in the `getCSSVariables` function:
```
    '--scrollbar-color' => 'var(--color-border-maxcontrast) transparent',
```

But then I would also reset this for high contrast mode:
And in `apps/theming/lib/Themes/HighContrastTheme.php`  `getCSSVariables` function:
```
    '--scrollbar-color' => 'auto',
```

Comment (Ornanovitch) [reply core/css/styles.scss:22]: Thanks!

I discovered there was already a `--color-scrollbar` variable that I reused and cleaned-up for this, tell me if its ok! It works well on my side, apart in the admin area where only the color is applied (the weight is still thin but I don't understand why).

File: apps/theming/lib/Themes/DarkHighContrastTheme.php [hunk 2]
```diff
@@ -89,8 +89,6 @@ public function getCSSVariables(): array {
 				'--color-info-hover' => $this->util->lighten($colorInfo, 10),
 				'--color-info-text' => $this->util->lighten($colorInfo, 20),
 
-				'--color-scrollbar' => $this->util->lighten($colorMainBackground, 35),
```
Comment (susnux) [inline apps/theming/lib/Themes/DarkHighContrastTheme.php:92]: Also set it to `auto` here.

File: apps/theming/css/default.css [hunk 3]
```diff
@@ -22,7 +22,6 @@
   --color-text-light: var(--color-main-text);
   /** @deprecated use `--color-text-maxcontrast` instead */
   --color-text-lighter: var(--color-text-maxcontrast);
-  --color-scrollbar: rgba(34,34,34, .15);
```
Comment (susnux) [inline apps/theming/css/default.css:25]: This needs to stay and be the same as in `DefaultTheme`:
`'--color-scrollbar' => 'var(--color-border-maxcontrast) transparent',`

File: core/css/styles.scss (hunk 1 above)
Comment (susnux) [reply core/css/styles.scss:22]: Few notes otherwise this is good :)

File: core/css/styles.scss [hunk 4]
```diff
@@ -155,28 +154,6 @@ body {
...
-
-::-webkit-scrollbar-corner {
-	background-color: transparent;
-}
-
-::-webkit-scrollbar-track-piece {
-	background-color: transparent;
-}
-
-::-webkit-scrollbar-thumb {
```
Comment (susnux) [inline core/css/styles.scss:173]: This is needed for Safari (not iOS), not sure if we want to drop it.

Comment (Ornanovitch) [reply core/css/styles.scss:173]: My bad! :pray:

**Review by susnux — APPROVED** [review]

**Review by jancborchardt — APPROVED** [review]
Awesome, thank you @Ornanovitch!

Btw, are you already in our Design room (link on https://nextcloud.com/design ) – I can also give you a guest account if you don't have one yet.

And are you aware of our Community Conference in Berlin? :) https://nextcloud.com/conf

**Comment by Ornanovitch** [comment]
The pleasure is all mine, I'd love to work on the Nextcloud design from time to time :)

> Btw, are you already in our Design room (link on https://nextcloud.com/design ) – I can also give you a guest account if you don't have one yet.

I'm not, let's go!

> And are you aware of our Community Conference in Berlin? :) https://nextcloud.com/conf

I'll check this out :)

Thanks again!

**Review by susnux — APPROVED** [review]

**Comment by susnux** [comment]
@Ornanovitch could you please run `npm run sass` locally and commit the files?

**Comment by Ornanovitch** [comment]
done!

**Review by marcoambrosini — APPROVED** [review]

**Comment by AndyScherzinger** [comment]
Nice work @Ornanovitch - sent you an invite to the Github organization of Nextcloud so you don't need to work via a fork for contributions. Should make life easier for you 👍 

Thanks for your work on the fix 💯

**Comment by jancborchardt** [comment]
> The pleasure is all mine, I'd love to work on the Nextcloud design from time to time :)

@Ornanovitch awesome, you should get an email with the details for the account. :) Added you to the "Design team" chat already.
And congrats, your contribution will be in the upcoming release in 2 weeks!

**Comment by skjnldsv** [comment]
This broke the high contrast sidebar :(
https://github.com/nextcloud/server/issues/49426

`auto` seems to not work on webkit browsers.

**Pull request merged**